messaging/src/main/java/io/lantern/messaging/metadata/Metadata.kt

```diff
@@ -17,16 +17,22 @@ import java.io.ByteArrayOutputStream
 import java.io.File
 import java.io.FileInputStream
 import java.io.IOException
-import mu.KotlinLogging
+import java.math.RoundingMode
 import kotlin.math.abs
 import kotlin.math.floor
+import mu.KotlinLogging
 
 private val logger = KotlinLogging.logger {}
 
 /**
  * Provides a facility for extracting content metadata while copying it
  */
-class Metadata(val mimeType: String?, val thumbnail: ByteArray?, val thumbnailMimeType: String?) {
+class Metadata(
+    val mimeType: String?,
+    val thumbnail: ByteArray?,
+    val thumbnailMimeType: String?,
+    val additionalMetadata: Map<String, String>? = null
```

Comment:
Additional metadata to use on the attachment just goes into a Map. Note that I default the map to null, so that we can ignore it when it doesn't apply.